src/garmin_parser.c

```diff
@@ -397,10 +398,19 @@ struct field_desc {
 	{ \
 		if (strcmp(#type, base_type_info[base_type].type_name)) \
 			fprintf(stderr, "%s: %s should be %s\n", #name, #type, base_type_info[base_type].type_name); \
-		type val = *(type *)p; \
+		type val; \
+		char is_endian_int = base_type_info[base_type].type_name[0] == 'S' || base_type_info[base_type].type_name[0] == 'U'; \
```

Comment:
Ugh. This is nasty. The 'S' will also trigger on 'STRING', and then we'll rely on that "type_size > 1" check to not do anything.

So it works, but..

I'd rather you do something like

     !memcmp(base_type_info[base_type].type_name+1, "INT")

but preferably in a helper inline function like "base_type_is_integer()" so that it would be a lot more legible.

Reply:
yeah, I like that better. I was struggling with the variable name - because that is what really shows how dumb this is (`STRING` isn't an endian int). I had noticed the issue, but figured with the `type_size > 1` check I was ok. But you are right. It's dumb. I'll fix it and move the whole thing into a helper.